src/ol_infrastructure/substructure/keycloak/__main__.py

```diff
@@ -224,24 +207,44 @@
     "ol-apps-update-password",
     realm_id=ol_apps_realm.realm,
     alias="UPDATE_PASSWORD",
-    default_action=True,
+    default_action=False,
     enabled=True,
     opts=resource_options,
 )
 
-ol_apps_touchstone_saml_identity_provider = keycloak.saml.IdentityProvider(
-    "touchstone-idp",
-    realm=ol_apps_realm.id,
-    alias="touchstone-idp",
-    display_name="MIT Touchstone",
-    entity_id=f"{keycloak_url}/realms/olapps",
-    force_authn=True,
-    post_binding_response=True,
-    principal_attribute="urn:oid:1.3.6.1.4.1.5923.1.1.1.6",
-    single_sign_on_service_url="https://idp.mit.edu/idp/profile/SAML2/Redirect/SSO",
-    trust_email=True,
-    validate_signature=True,
-    want_assertions_encrypted=True,
-    want_assertions_signed=True,
-    opts=resource_options,
-)
+
+# Check if any Openid clients exist in config and create them
+for openid_clients in keycloak_config.get_object("openid_clients"):
+    realm_name = openid_clients.get("realm_name")
+    client_details = openid_clients.get("client_info")
+    for client_name, client_url in client_details.items():
+        openid_client = keycloak.openid.Client(
+            f"ol-{client_name}-client",
+            realm_id=realm_name,
+            client_id=f"ol-{client_name}-client",
+            enabled=True,
+            access_type="CONFIDENTIAL",
+            standard_flow_enabled=True,
+            valid_redirect_uris=[f"{client_url}/*"],
+            login_theme="keycloak",
+            opts=resource_options,
+        )
+
+if stack_info.env_suffix != "ci":
+    ol_apps_touchstone_saml_identity_provider = keycloak.saml.IdentityProvider(
```

Comment:
Can/should we also enable touchstone on the platform realm?

Reply:
Probably not as we either want to create the developer accounts ourselves or integrate with Github and write some code to adjust the action in the login flow to check for org membership